File: frontend/src/types/inventory.ts
export interface InventoryItem {
  asset_id: string
  class_id: string
  instance_id: string
  app_id: number
  name: string
  icon_url: string
  marketable: boolean
  tradable: boolean
  type: string
  rarity: string
  rarity_color: string
  exterior: string
  account_index: number
  price_usd: number | null
}

export interface InventoryResult {
  app_id: number
  total_items: number
  items: InventoryItem[]
  total_value_usd: number | null
  priced_item_count: number
}

export interface SupportedGame {
  app_id: number
  name: string
}

File: backend/app/schemas/inventory.py
from pydantic import BaseModel


class InventoryAccountInput(BaseModel):
    steam_ids: list[str]
    app_id: int
    include_prices: bool = False


class InventoryItem(BaseModel):
    asset_id: str
    class_id: str
    instance_id: str
    app_id: int
    name: str
    icon_url: str
    marketable: bool
    tradable: bool
    type: str
    rarity: str
    rarity_color: str
    exterior: str
    account_index: int
    price_usd: float | None = None


class InventoryResult(BaseModel):
    app_id: int
    total_items: int
    items: list[InventoryItem]
    total_value_usd: float | None = None
    priced_item_count: int = 0